packages/form-layout/src/vaadin-form-item.js

```diff
@@ -166,12 +166,71 @@ class FormItem extends ThemableMixin(PolymerElement) {
     this.__contentFieldObserver = new MutationObserver(() => this.__updateRequiredState(this.__contentField.required));
   }
 
+  /** @protected */
+  ready() {
+    super.ready();
+
+    // Ensure every instance has unique ID
+    const uniqueId = (FormItem._uniqueLabelId = 1 + FormItem._uniqueLabelId || 0);
+    this.__labelId = `label-${this.localName}-${uniqueId}`;
+
+    if (this.__labelNode) {
+      this.__labelNode.id = this.__labelId;
+      this.__linkLabelToField(this.__labelNode);
```

Comment:
This is unnecessary because the initial "slotchange" also triggers `__linkLabelToField`

Reply:
Fixed.